server/apps/store/subscriptions_test.go

```diff
@@ -0,0 +1,353 @@
+		mockAPI.On("KVGet", subKey).Return(nil, model.NewAppError("KVGet", "test", map[string]interface{}{}, "test error", 0)).Times(1)
+		err := s.StoreSub(&toStore)
+		require.Error(t, err)
+		require.Equal(t, "KVGet: test, test error", err.Error())
+	})
+
+	t.Run("no value for subs key", func(t *testing.T) {
+		mockAPI.On("KVGet", subKey).Return(nil, nil).Times(1)
+		mockAPI.On("KVSetWithOptions", subKey, emptySubsWithToStoreBytes, mock.Anything).Return(true, nil).Times(1)
+		err := s.StoreSub(&toStore)
+		require.NoError(t, err)
+	})
+
+	t.Run("empty list for subs key", func(t *testing.T) {
+		mockAPI.On("KVGet", subKey).Return(emptySubsBytes, nil).Times(1)
+		mockAPI.On("KVSetWithOptions", subKey, emptySubsWithToStoreBytes, mock.Anything).Return(true, nil).Times(1)
+		err := s.StoreSub(&toStore)
+		require.NoError(t, err)
+	})
+
+	t.Run("error setting subscription", func(t *testing.T) {
+		mockAPI.On("KVGet", subKey).Return(storedSubsBytes, nil).Times(1)
+		mockAPI.On("KVSetWithOptions", subKey, storedSubsWithToStoreBytes, mock.Anything).Return(false, model.NewAppError("KVSet", "test", map[string]interface{}{}, "test error", 0)).Times(1)
+		err := s.StoreSub(&toStore)
+		require.Error(t, err)
+		require.Equal(t, "KVSet: test, test error", err.Error())
+	})
+
+	t.Run("subscription stored", func(t *testing.T) {
+		mockAPI.On("KVGet", subKey).Return(storedSubsBytes, nil).Times(1)
+		mockAPI.On("KVSetWithOptions", subKey, storedSubsWithToStoreBytes, mock.Anything).Return(true, nil).Times(1)
+		err := s.StoreSub(&toStore)
+		require.NoError(t, err)
+	})
+}
+
+func TestSubsKey(t *testing.T) {
+	s := store{}
+	for name, testcase := range map[string]struct {
+		Subject   apps.Subject
+		TeamID    string
+		ChannelID string
+		Expected  string
+	}{
+		string(apps.SubjectUserCreated): {
+			apps.SubjectUserCreated,
+			"team-id",
+			"channel-id",
+			"sub_user_created",
+		},
+		string(apps.SubjectUserJoinedChannel): {
+			apps.SubjectUserJoinedChannel,
+			"team-id",
+			"channel-id",
+			"sub_user_joined_channel_channel-id",
+		},
+		string(apps.SubjectUserLeftChannel): {
+			apps.SubjectUserLeftChannel,
+			"team-id",
+			"channel-id",
+			"sub_user_left_channel_channel-id",
+		},
+		string(apps.SubjectUserJoinedTeam): {
+			apps.SubjectUserJoinedTeam,
+			"team-id",
+			"channel-id",
+			"sub_user_joined_team_team-id",
+		},
+		string(apps.SubjectUserLeftTeam): {
+			apps.SubjectUserLeftTeam,
+			"team-id",
+			"channel-id",
+			"sub_user_left_team_team-id",
+		},
+		string(apps.SubjectUserUpdated): {
+			apps.SubjectUserUpdated,
+			"team-id",
+			"channel-id",
+			"sub_user_updated",
+		},
+		string(apps.SubjectChannelCreated): {
+			apps.SubjectChannelCreated,
+			"team-id",
+			"channel-id",
+			"sub_channel_created_team-id",
+		},
+		string(apps.SubjectPostCreated): {
+			apps.SubjectPostCreated,
+			"team-id",
+			"channel-id",
+			"sub_post_created_channel-id",
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
```

Comment:
I think you'll have to do this as well if you make the other change

```suggestion
		t.Run(string(name), func(t *testing.T) {
```